dangerzone/isolation_provider/base.py

```diff
@@ -209,6 +212,35 @@ def terminate_doc_to_pixels_proc(
         """Terminate gracefully the process started for the doc-to-pixels phase."""
         pass
 
+    def _signal_process_group(self, p: subprocess.Popen, signo: int) -> None:
```

Comment:
Yeah, we do have a `util.py`, and we could move it there, but there are a few assumptions that are tied with the isolation provider:

1. The fact that we don't raise an error if the process no longer exists. In a neutral setting, we would have to inform the caller about it, but it doesn't make sense for the doc-to-pixels process.
2. The fact that we don't kill the process group on Windows, and we fallback to killing the process. That's a choice that makes sense for the doc-to-pixels process, but I'm not sure we can generalize it :-/

Reply:
Thanks for the explanations.

If you don't think it's worth generalizing, I suggest removing it from the class and having it as a first level function of the python module rather than being a method.

If we were to generalize it, naming would be important, and parameters to these methods might also be useful (like `fallback_to_process` and `raise_on_error`, for instance) so that it's clear what it's done.

Reply:
Ok, then let's go with turning the methods into module-level functions: 0d55aea63552c4bfdd079348730b473374bd559f